MovieChecker/MovieCheckerTests/MovieCheckerTests.m

```diff
@@ -6,35 +6,118 @@
+
 @implementation MovieCheckerTests
 
-- (void)setUp {
++ (void)setUp
+{
     [super setUp];
-    // Put setup code here. This method is called before the invocation of each test method in the class.
+    
+    // Register the mock URL protocol class, so that all calls will be mocked, rather than hitting a remote server.
+    [NSURLProtocol registerClass:[VOKMockUrlProtocol class]];
 }
 
-- (void)tearDown {
-    // Put teardown code here. This method is called after the invocation of each test method in the class.
++ (void)tearDown
+{
+    // Un-register the mock URL protocol class.
+    [NSURLProtocol unregisterClass:[VOKMockUrlProtocol class]];
+    
     [super tearDown];
 }
 
-- (void)testExample {
-    // This is an example of a functional test case.
-    XCTAssert(YES, @"Pass");
+- (void)setUp
+{
+    [super setUp];
+    // Put setup code here. This method is called before the invocation of each test method in the class.
+    self.session = [NSURLSession sessionWithConfiguration:[NSURLSessionConfiguration ephemeralSessionConfiguration]];
+    Class mockURLProtocol = [VOKMockUrlProtocol class];
+    NSMutableArray *currentProtocolClasses = [self.session.configuration.protocolClasses mutableCopy];
+    [currentProtocolClasses insertObject:mockURLProtocol atIndex:0];
+    self.session.configuration.protocolClasses = [currentProtocolClasses copy];
+    
 }
 
-- (void)testPerformanceExample {
-    // This is an example of a performance test case.
-    [self measureBlock:^{
-        // Put the code you want to measure the time of here.
-    }];
+- (void)testDownloadBoxOfficeList
+{
+    BOOL __block done = NO;
+    NSURL *url = [RottenTomatoesHelperMethods boxOfficeURL];
+    NSURLSessionDataTask *dataTask = [self.session
+                                      dataTaskWithURL:url
+                                      completionHandler:^(NSData *data, NSURLResponse *response, NSError *error) {
+                                          XCTAssertNil(error, @"Got an error: %@", error);
+                                          NSHTTPURLResponse *httpResp = (NSHTTPURLResponse *)response;
+                                          XCTAssertEqual(httpResp.statusCode, 200, @"Didn't get the expected 200 status.");
+                                          NSArray *movies = [RottenTomatoesHelperMethods
+                                                             interpretBoxOfficeMoviesFromData:data withkey:@"movies"];
```

Comment:
It looks like this method tests `+ [RottenTomatoesHelperMethods boxOfficeURL]` and `+ [RottenTomatoesHelperMethods interpretBoxOfficeMoviesFromData:]`, but nothing from `MovieList`, which is the class that does the actual list-loading in your app.